client/components/app.js

```diff
@@ -25,57 +23,44 @@ class App extends Component {
   }
 
   onClickPlanTrip() {
-    this.setState({ clickPlan: !this.state.clickPlan });
+    this.setState((prevState) => ({ clickPlan: !prevState.clickPlan }));
   }
 
   onClickGetTrips() {
```

Comment:
use arrow function here to keep this binding, otherwise you probaby need to bind